NimbleMedium/Sources/Presentation/Modules/Feeds/FeedsView.swift

```diff
@@ -71,56 +68,96 @@ struct FeedsView: View {
+
+        let viewModel: FeedsViewModelProtocol
+
+        @State var isRefeshing: Bool = false
+        @State var hasMore: Bool = true
+        @State var isLoadingMore: Bool = false
+        @State var feedRowViewModels: [FeedRowViewModelProtocol] = []
+        @State var isPushDetailActive = false
+        @State var activeDetailID = ""
+
+        var body: some View {
+            Group {
+                if !feedRowViewModels.isEmpty {
+                    ScrollView {
+                        feedRows
+                    }
+                    .enableRefresh()
+                    .padding(.top, 16.0)
+                } else {
+                    Text(Localizable.feedsNoArticle())
                 }
             }
-            .padding(.all, 16.0)
-            .frame(maxWidth: .infinity, alignment: .leading)
-
-            if hasMore {
-                RefreshFooter(
-                    refreshing: $isLoadingMore,
-                    action: { viewModel.input.loadMore() },
-                    label: { ProgressView() }
-                )
+            .onReceive(viewModel.output.didFinishRefresh) { _ in
+                isRefeshing = false
             }
+            .onReceive(viewModel.output.didFinishLoadMore) {
+                hasMore = $0
+                isLoadingMore = false
+            }
+            .bind(viewModel.output.feedRowViewModels, to: _feedRowViewModels)
         }
-    }
 
-    var feedList: some View {
-        Group {
-            if !feedRowViewModels.isEmpty {
-                ScrollView {
-                    feedRows
+        var feedRows: some View {
+            Group {
+                detailNavigationLink
+                RefreshHeader(
+                    refreshing: $isRefeshing,
+                    action: { viewModel.input.refresh() },
+                    label: { _ in ProgressView() }
+                )
+
+                // FIXME: LazyVStack produces an infinity refresh FeedRow
+                ForEach(feedRowViewModels, id: \.output.id) { viewModel in
+                    FeedRow(viewModel: viewModel)
+                        .padding(.bottom, 16.0)
+                        .onTapGesture {
+                            activeDetailID = viewModel.output.id
+                            isPushDetailActive = true
+                        }
+                }
+                .padding(.all, 16.0)
+                .frame(maxWidth: .infinity, alignment: .leading)
+
+                if !isPushDetailActive && hasMore {
+                    RefreshFooter(
+                        refreshing: $isLoadingMore,
+                        action: { viewModel.input.loadMore() },
+                        label: { ProgressView() }
+                    )
                 }
-                .enableRefresh()
-                .padding(.top, 16.0)
-            } else {
-                Text(Localizable.feedsNoArticle())
             }
         }
+
+        var detailNavigationLink: some View {
```

Comment:
maybe we can update the name a bit for better clarification:
```suggestion
        var feedDetailNavigationLink: some View {
```